technical/validations.md

```diff
@@ -7,20 +7,49 @@ these validations will be duplicated in the front-end code for an optimal user e
+- at least 4 decimal precisions required for longitude and latitude of the playground location, in order to avoid duplicate entries
+- a user can join the playground where he is already a volunteer in, then system just returns success and ignore the request
+- the id, name, lat, lng and status are required properties during creation of a new playground
+- for onboarding smokefree playground, the smokeFreeDate must be past
+- smokefreeDate can only be reset by managers, if the smokefreeDate is passed then it can not be reset
+- the status can not be stopped unless the playground managers choose that the playground can not be smokefree
+- the type can not be smokefree unless it get consecutive 500 smokefree playground observations and status must be finished
+- the status can not be finished unless all the phases and actions are completed
 
 ### Playground workspace
-- The maximum number of volunteers that can enter a playground workspace is 200
-- The maximum number of managers in a workspace is ..
+- the maximum number of volunteers that can enter a playground workspace is 200
+- the maximum number of managers in a workspace is volunteers/5
+- only playground managers can decide whether playground can be smokefree or not
+- the playground can not be decided if status is not_started or stopped
+- only playground managers can set smokefreeDate and they cannot set smokefreeDate unless they decide playground to be smokefree
+- managers will not be able to spread/share the date unless smokefreeDate is chosen
+- the last playground manager can not unsubscribe without choosing one from the recommended list of volunteers
+- if a volunteer  claims the manager role where he is already a manager , then system return success by ignoring the request
+
 
 
 ### Chat message
 - a chat message cannot consist of whitespace only
 - the minimum length of a chat message is 1 characters
 - the maximum length of a chat message is 300 characters
-- this is the set of valid characters for a chat message: ...  (or provide a regexp)
+- this is the set of valid characters for a chat message: [!@&(),.?":]
+
+### Playground Observation
+- a comment message can't consist of whitespaces only
+- no whitespaces in the beginning or/and trailing of a comment
+- the minimum length of a comment message is 100 characters
+- the maximum length of a comment message is 2000 characters
+- how frequently a volunteer can indicate his observations about the playground
+- the indicated playground observation will be shown after verification of the comment by anyone of the managers
+- consecutive 500 smokefree playground observations considered that The playground is now smokefree
+- the maximum number of playground observations for the playground can be stored are 1000
+- the smokefree indication, observer, initiativeId are required and no default value is set
+- multiple volunteers can not indicate observation for a particular playground at the same time
```

Comment:
I understand, but just only do the validation of one observation per volunteer per playground per day